### ISSUE TITLE
Ensure renaming operations are consistent across the product

### ISSUE THREAD

**Issue by kgodey** [mathesar-foundation/mathesar#1682]
We need to make sure that we're using the same UX for renaming:
- Schemas
- Tables
- Columns
- Explorations

Blocked by
- #1659 
- #1660 
- #1664
- #1662 
- #1663

**Comment by kgodey** [comment]
Closing this, too old, requirements are likely out of date. We can create a new issue if we need this functionality in the future.
